### DISCUSSION TITLE
I'm trying to connect myphone for a initial test but i can't. Could you help me?

### DISCUSSION THREAD

**Discussion by devPedroFreire** [Auties00/WhatsappWeb4j#359] [General]
def conectarWA(){

        def conn = Whatsapp.webBuilder()
        def newConn = conn.newConnection(999999999999).unregistered(QrHandler.toTerminal())

    }

how can I connect via QRcode?

**Comment by Auties00** [comment]
>     def conectarWA(){
> 
>         def conn = Whatsapp.webBuilder()
>         def newConn = conn.newConnection(5584981578606).unregistered(QrHandler.toTerminal())
> 
>     }
> 
> 
> how can I connect via QRcode? 

are you using groovy? Also you are missing 
.connect().await()

**Reply by devPedroFreire** [reply]
![image](https://github.com/Auties00/Cobalt/assets/39525797/1413cf59-0e23-46f4-93fe-eed2920a789b)


Im using groovy. This is the latest version, would you recommend another?

**Reply by devPedroFreire** [reply]
in the image i didnt put '()' after the connect, but im having an error either way

**Reply by devPedroFreire** [reply]
could you give an exemple?

**Reply by Auties00** [reply]
can you show me the error?
This should work:
Whatsapp.webBuilder().newConnection(5584981578606).unregistered(QrHandler.toTerminal()).connect().awaitDisconnection()

**Reply by devPedroFreire** [reply]
how can i get the instance of my connection to send message or list my contacts??

**Reply by devPedroFreire** [reply]
i already see the qrcode on terminal, but what do i do after a read it? its not clear to me on READ.ME

**Reply by devPedroFreire** [reply]
how can I do a listener to list all chats? i really didnt find out how to do it
